ophyd/status.py

```diff
@@ -77,11 +77,15 @@ def _wait_and_cleanup(self):
                 timeout = None
             wait(self, timeout=timeout, poll_rate=0.2)
         except TimeoutError:
-            logger.debug('Status object %s timed out', str(self))
-            try:
-                self._handle_failure()
-            finally:
-                self._finished(success=False)
+            with self._lock:
+                if self.done:
+                    # Avoid race condition with settling.
+                    return
+                logger.debug('Status object %s timed out', str(self))
```

Comment:
Does all of this need to be in the lock?

Reply:
Yes. We can't have settling start running while failure is being handled.

Reply:
right, because we set `done` pretty late

Reply:
Exactly